pkg/reconciler/tenancy/initialization/apibinder_initializer_controller.go

```diff
@@ -0,0 +1,304 @@
+			c.enqueueClusterWorkspace(obj, logger, "")
+		},
+	})
+
+	apiBindingsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			c.enqueueAPIBinding(obj, logger)
+
+		},
+		UpdateFunc: func(_, obj interface{}) {
+			c.enqueueAPIBinding(obj, logger)
+		},
+	})
+
+	clusterWorkspaceTypeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			c.enqueueClusterWorkspaceType(obj, logger)
+		},
+		UpdateFunc: func(_, obj interface{}) {
+			c.enqueueClusterWorkspaceType(obj, logger)
+		},
+	})
+
+	return c, nil
+}
+
+type clusterWorkspaceResource = committer.Resource[*tenancyv1alpha1.ClusterWorkspaceSpec, *tenancyv1alpha1.ClusterWorkspaceStatus]
+
+// APIBinder is a controller which instantiates APIBindings and waits for them to be fully bound
+// in new ClusterWorkspaces.
+type APIBinder struct {
+	queue workqueue.RateLimitingInterface
+
+	getClusterWorkspace     func(clusterName logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error)
+	getClusterWorkspaceType func(clusterName logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspaceType, error)
+	listClusterWorkspaces   func() ([]*tenancyv1alpha1.ClusterWorkspace, error)
+
+	listAPIBindings  func(clusterName logicalcluster.Name) ([]*apisv1alpha1.APIBinding, error)
+	createAPIBinding func(ctx context.Context, clusterName logicalcluster.Name, binding *apisv1alpha1.APIBinding) (*apisv1alpha1.APIBinding, error)
+	deleteAPIBinding func(ctx context.Context, clusterName logicalcluster.Name, bindingName string) error
+
+	createdBindingsLock sync.Mutex
+	createdBindings     map[logicalcluster.Name]map[tenancyv1alpha1.APIExportReference]struct{}
+
+	getAPIExport func(clusterName logicalcluster.Name, name string) (*apisv1alpha1.APIExport, error)
+
+	transitiveTypeResolver *admission.TransitiveTypeResolver
+
+	// commit creates a patch and submits it, if needed.
+	commit func(ctx context.Context, new, old *clusterWorkspaceResource) error
+}
+
+func (b *APIBinder) enqueueClusterWorkspace(obj interface{}, logger logr.Logger, logSuffix string) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	logging.WithQueueKey(logger, key).V(2).Info(fmt.Sprintf("queueing ClusterWorkspace%s", logSuffix))
+	b.queue.Add(key)
+}
+
+func (b *APIBinder) enqueueAPIBinding(obj interface{}, logger logr.Logger) {
+	metaObj, ok := obj.(metav1.Object)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("obj is supposed to be a metav1.Object, but is %T", obj))
+		return
+	}
+	clusterName := logicalcluster.From(metaObj)
+
+	clusterWorkspace, err := b.getClusterWorkspace(clusterName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// The workspace was deleted or is no longer initializing, so we can safely ignore this event.
+			return
+		}
+
+		parentClusterName, clusterWorkspaceName := clusters.SplitClusterAwareKey(clusterName.String())
```

Comment:
`clusterName` is a `logicalcluster.Name{}` - what does it mean to call `SplitClusterAwareKey` on it?

Reply:
will revise